app/components/Blockchain/Transaction/Transaction.jsx

```diff
@@ -174,7 +185,7 @@ export default class Transaction extends React.Component<Props> {
   renderAbstractN3 = () => {
     const { isPending, tx, address } = this.props
     const { time, type, sender } = tx
-    const txDate = this.renderTxDate(time || tx.metadata.time)
+    const txDate = this.renderTxDate(time || (tx.metadata && tx.metadata.time))
```

Comment:
This is the type error that was causing `v2.12.3` to crash... The rest of this PR just adds enhancements and fixes a few other issues I noticed 😅